include/swift-c/SyntaxParser/SwiftSyntaxParser.h

```diff
@@ -232,6 +232,65 @@ swiftparse_parse_string(swiftparse_parser_t, const char *source);
 /// declarations, etc.
 SWIFTPARSE_PUBLIC const char* swiftparse_syntax_structure_versioning_identifier(void);
 
+typedef struct {
+  /// Represents the range for the fixit.
+  swiftparse_range_t range;
+  /// Represent the text for replacement.
+  const char* text;
+} swiftparse_fixit_t;
+
+enum swiftparser_diagnostic_severity {
+  swiftparser_diagnostic_severity_Error = 0,
+  swiftparser_diagnostic_severity_Warning = 1,
+  swiftparser_diagnostic_severity_Remark = 2,
```

Comment:
I don't think "remark" diagnostics are relevant for the parser, these are used for things like optimizer hints. I think you can ignore this, and just add an assertion on the libSwiftSyntaxParser side that we don't ever receive such kind of diagnostic from the parser.